### ISSUE TITLE
Chrome Out Of Memory Issue

### ISSUE THREAD

**Issue by DanielDarrenJones** [statamic/cms#9351]
### Bug description

We have had a report from one of our clients about an out-of-memory crash in Chrome when editing entries, we have been having other issues with crashes See (https://github.com/statamic/cms/issues/9299), but this seems to potentially be a separate issue after talking to @duncanmcclean. Below is the report from the one of our clients team with the context of the conversation being a continuation of trying to track down the cause of the crashes we reported previously:

It could have something to do with how much memory Statamic is using up when actively using it. I have two tabs open, the one I have actually been editing on is using 2.1 GB of memory usage, which seems like a huge amount. The next lowest is Canva at 650MB 9the document I have open is 275 tabs of images, the max number of tabs you can have is 300), and all other tabs are between 200MB-30MB.

![image](https://github.com/statamic/cms/assets/8768709/37403a5b-3e60-4621-ba2d-ad713bac70be)

For reference, all I have done on this page so far is:
Change page title
Change page thumbnail
Edit Body Block – update hero image from asset library and change word Indonesia to Papua New Guinea. 
Edit Breadcrumbs, changing text, and amending one link
Editing some text in the intro body block
 
It is a very lean page compared to some, so for it be using such a high memory seems concerning and could be the root of the issue on both Chrome & Safari. 
 
My Chrome crashed yesterday, but I didn’t have time to capture everything before it refreshed which is why I didn’t send it over, however I did manage to capture this on the web dev tools:

![image](https://github.com/statamic/cms/assets/8768709/3ff90906-77c4-495f-a18c-54d4d91e2dc1)

I checked my memory of all my Google Accounts I am logged into on this device, all accounts have way over 10GB of memory left so it wouldn’t be that slowing down. 
 
Here is the other Statamic’s tab memory usage, one that I had not made any edits on.

![image](https://github.com/statamic/cms/assets/8768709/e8dffcac-d9d1-40bc-abe8-edf1a0fc346b)

### How to reproduce

I haven't been able to reproduce this reliably, the client has been using two tabs, one with the asset page on to upload and manage assets, and another to create/duplicate as well as edit entries. I think this is leading to issues, but its really hard to track this down.

Our pages make use of the replicated quite heavily for a page builder style setup where each block can be customised, I believe they also make use of the live preview quite a lot.

### Environment

```yaml
Environment
Application Name: Pelorus
Laravel Version: 10.40.0
PHP Version: 8.2.12
Composer Version: 2.6.6
Environment: production
Debug Mode: OFF
URL: pelorus-travel.strattons.com/
Maintenance Mode: OFF

Cache
Config: CACHED
Events: NOT CACHED
Routes: CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: redis
Database: mysql
Logs: stack / single
Mail: smtp
Queue: redis
Session: redis

Statamic
Addons: 6
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.44.0 PRO

Statamic Addons
doefom/currency-fieldtype: 1.2.3
gerttimmerman/statamic-zapier: 1.5
jacksleight/statamic-bard-texstyle: 3.1.4
ndx/statamic-bard-color-picker: 1.2.0
ryanmitchell/statamic-translation-manager: 1.3.0
withcandour/aardvark-seo: 3.0.0
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

Runtime (default)

**Comment by duncanmcclean** [comment]
We released some performance improvements relating to Bard & Replicator yesterday in [v4.45.0](https://github.com/statamic/cms/releases/tag/v4.45.0) (PR: #9353). It might be worth updating to see if that helps in terms of memory usage.

**Comment by jasonvarga** [comment]
Going to assume the PR mentioned solved this, but feel free to let us know if this is still an issue and we'd be happy to reopen this.

**Comment by DanielDarrenJones** [comment]
@duncanmcclean  @jasonvarga We have just started running into this again, I have managed to replicate the memory usage on the Chrome tab spiking, it generally seems to happen when we are selecting a few images on the page across multiple blocks.

We are using a replicator which has 22 blocks in it, with a total of 10 images on the page, other members of the team have had crashes again, but I haven't been able to get it to crash yet, however, the memory on the tab spiked to over 1GB in my testing.

Here is a video showing the issue: https://we.tl/t-kwznttUGUG

**Comment by jasonvarga** [comment]
There's just so many fields / vue components on the page at one time. You have 22 blocks, but each block has multiple fields, including nested replicators/bards/grids which have even more fields.

There's currently no magic bullet to solve this at the moment aside from avoiding a gigantic replicator field.

The way you could do that is to re-think your replicator sets into entries:
- Create a "page components" collection
- Each replicator set becomes a blueprint in the page components collection
- Your replicator field becomes an "entries" field
- The "enabled" toggles are replaced by marking the entries as drafts.
- Instead of clicking "add set" and picking a set, you click "create entry" and pick a blueprint.

The benefit of this:
- There's hardly anything on the page at one time, which manages the memory much better. The set/entry opens in a stack and when you're done, it closes.

The cons of this:
- You might find the huge component collection to be unmanageable.
- Since the sets are entries, as you work on them, they become immediately published, even before you save the parent page.

I'm working on a demo site for this approach, and possibly a command or some other way to migrate from an existing replicator to entries since obviously it would be tedious to do by hand.

**Comment by DanielDarrenJones** [comment]
@jasonvarga It's weird that the problem is only surfacing with assets though - there has to be something here that's causing it, especially with it jumping up so dramatically.

**Comment by jasonvarga** [comment]
Okay cool if it's definitely the assets thats not letting memory free up that's really helpful. Thank you!

**Comment by DanielDarrenJones** [comment]
As far as I can tell it seems to be stemming from the asset browser. I have some heap dumps from Chrome with the different memory usages and breakdowns on fresh load, clearing the existing assets, and then after I have searched and selected new assets. Not sure how useful this would be though as they are very verbose and most of the function calls are low-level Vue reactivity functions.

The issues stemming from the browser issue would also line up with what our client said, if they just upload new assets it's fine, it's only when they go to select an existing one.

Please let me know if there is absolutely anything else I can provide to help track this down further - really keen to get this resolved.

**Comment by DanielDarrenJones** [comment]
@jasonvarga another interesting thing - reloading the page seems to jump the memory usage - this climbs as you load more and more pages, but never seems to drop back down to the same amount of memory as when the page was loaded fresh initially - is there something being cached / stored? https://www.loom.com/share/da572308d0524231920e7f373601d3f0

**Comment by jasonvarga** [comment]
That seems to be true of any website. You can go to google.com with the memory profiler open and refresh over and over watching the memory increase like that.

**Comment by DanielDarrenJones** [comment]
> That seems to be true of any website. You can go to google.com with the memory profiler open and refresh over and over watching the memory increase like that.

Sorry! Easy to fall down a rabbit hole with this stuff, thank you so much for taking the time to look into it.

**Comment by jasonvarga** [comment]
Can I close this issue now after #9500 or is it still crawling?

**Comment by DanielDarrenJones** [comment]
Let me check in with the client on Tuesday and see - you can likely close this for the time being though. Thank you!

**Comment by jasonvarga** [comment]
Ok thanks, let us know. 👍
